src/accelerate/utils/modeling.py

```diff
@@ -1351,7 +1355,12 @@ def load_state_dict(checkpoint_file, device_map=None):
             else:
                 progress_bar = None
             for device in devices:
-                with safe_open(checkpoint_file, framework="pt", device=device) as f:
+                target_device = device
+
+                if is_xpu_available() and isinstance(device, int):
+                    target_device = f"xpu:{device}"
+
+                with safe_open(checkpoint_file, framework="pt", device=target_device) as f:
```

Comment:
Just to be sure, the model won't be loaded in the correct xpu device if we don't have the prefix "xpu:". Is that right ? cc @statelesshz since you added the support for npu and it might also be required.

Reply:
Yes, you can see it here: https://github.com/huggingface/safetensors/pull/428 file `bindings/python/src/lib.rs` line 263:

```rust
 name if name.starts_with("xpu:") => {
    let tokens: Vec<_> = name.split(':').collect();
    if tokens.len() == 2 {
        let device: usize = tokens[1].parse()?;
        Ok(Device::Xpu(device))
    } else {
        Err(SafetensorError::new_err(format!(
            "device {name} is invalid"
        )))
    }
}
```

If you supply only a number, it will treat as CUDA device (the same file, line 278):

```rust
} else if let Ok(number) = ob.extract::<usize>() {
            Ok(Device::Cuda(number))
}
```

Reply:
Yes I think the changes are correct . It requests a "xpu" device ; since the addition of safetensors.

Reply:
Perfect ! Thanks for the explanation !